deploy_docs.sh

```diff
@@ -0,0 +1,59 @@
+# Modified from https://github.com/AMReX-Codes/amrex/blob/master/build_and_deploy.sh
+# and licensed according to BSD-3-Clause-LBNL (https://github.com/AMReX-Codes/amrex/blob/master/LICENSE).
+
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# Build the documentation from the SOURCE_BRANCH
+# and push it to TARGET_BRANCH.
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deply)
+git clone $REPO out
+cd out
+
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+cd ..
+
+# Clean out existing contents
+rm -rf out/sphinx_docs/ || exit 0
+
+# Pull from SOURCE_BRANCH again
+git pull $SSH_REPO $SOURCE_BRANCH
+
+# Build the Sphinx documentation
+cd docs
+make html
+cd ../  
+
+mkdir -p out/sphinx_docs/
+mv -f docs/_build/html/* out/sphinx_docs/
```

Comment:
Hence, this will also be required to change. Except this, your entire code is absolutely correct (even originating source for my code).

I suggest to replace your code from line 23 to 37, with following:

```

# Clean out existing contents
git rm -rf . || exit 0
cd ..

# Build the Sphinx documentation
cd docs
make html
cd ..

# Move built docs to out/
mv -f docs/_build/html/* out
```

Try this, and please let me know if there are any problems. I hope it should work fine - in that case, please share your gh-pages branch.

Reply:
I am having trouble implementing this before judging it. If I start line 23 with git rm -rf. , I remove all the contents of my repository, including the files needed for sphinx. Then I cd out of the main source repository all together, so I cannot apply the other commands, such as cd docs, which has no index file anymore anyway.

Reply:
I think you got confused by the way I told you to replace from 23-37, I meant including these lines 23 & 37. This means you are not including `cd ..` before putting my code. You are executing `git rm -rf .` within `out/` dir - so it will clean gh-pages branch not your repository and thereby you won't get out of your source repository in next `cd ..`

Please feel free to let me know if I'm unable to understand your trouble.

Reply:
So you think  it is unnecessary to have anything in gh-pages other than the documentation itself? That sounds reasonable to me. So, we would delete all the actual tardis code, correct ?

Reply:
Yes, exactly. gh-pages branch are meant to be so. We will only put *built docs* (i.e. html, rst.txt, js, etc. files) generated by sphinx, in there in root directory.

Yes for 1st commit it would appear to delete all TARDIS code mistakenly present already there.

Reply:
Also for my suggested changes I created a PR to avoid confusion: https://github.com/Youssef15015/tardis/pull/3